api/graphql/schema.graphql

```diff
@@ -1672,6 +1696,11 @@ type Mutation {
     @delete
     @guard
     @can(ability: "delete", find: "id")
+  updateUserTeamRoles(
+    teamId: ID!
+    userId: UUID!
+    teamRoleAssignments: UpdateUserTeamRolesInput! @spread
```

Comment:
I went back and forth on this, but its based on an idea I read [here](https://github.com/graphql-rules/graphql-rules/blob/master/docs/rules/06-mutations/mutation-input-arg.md). The idea is that by putting input fields in a nested object, it's easier to work with and modify. You can't deprecate top-level arguments, but you can deprecate _fields_ in an argument object.

Reply:
BTW @petertgiles the same set of reccomended best practices inspired my recent talk of more focused mutations: https://github.com/graphql-rules/graphql-rules/blob/master/docs/rules/06-mutations/mutation-business-operations.md

Reply:
Hmm, that was an interesting read.  You got stuck half-way though - you can't move teamId and userId into your [single mutation parameter](https://github.com/graphql-rules/graphql-rules/blob/master/docs/rules/06-mutations/mutation-input-arg.md) because of the can directive?  If you could move it in that might help alleviate the weirdness of having a team input with no team ID.

Reply:
Hmm, after this last change, I think I like this pattern!